test/integration/oauth_server_headers_test.go

```diff
@@ -68,18 +70,35 @@ func checkNewReqHeaders(t *testing.T, rt http.RoundTripper, check_url string) {
 		t.Fatalf("unexpected error %v", err)
 	}
 
-	checkImportantHeaders := map[string]string{
+	importantHeaders := http.Header{}
+	for key, val := range map[string]string{
+		// security related headers that we really care about, should not change
+		"Cache-Control":          "no-cache, no-store",
+		"Pragma":                 "no-cache",
+		"Expires":                "0",
 		"Referrer-Policy":        "strict-origin-when-cross-origin",
 		"X-Frame-Options":        "DENY",
 		"X-Content-Type-Options": "nosniff",
 		"X-DNS-Prefetch-Control": "off",
 		"X-XSS-Protection":       "1; mode=block",
+
+		// non-security headers, should not change
+		// adding items here should be validated to make sure they do not conflict with any security headers
+		// "KEY": "VAL",  // stub placeholder for future values
+	} {
+		// use set so we get the canonical form of these headers
+		importantHeaders.Set(key, val)
+	}
+
+	// these headers can change per request and are not important to us
+	// only add items to this list if they cannot be statically checked above
+	ignoredHeaders := []string{"Date", "Content-Length", "Content-Type", "Location"}
```

Comment:
content-type does not change at a whim and should be checked too.
all the pages we check are HTML pages so we should get a consistent value here.